manticore/core/cpu/abstractcpu.py

```diff
@@ -756,10 +756,11 @@ def fallback_to_emulate(*operands):
             for l in self.render_registers():
                 register_logger.debug(l)
 
-        implementation(*insn.operands)
-        self._icount += 1
-
-        self._publish('did_execute_instruction', self._last_pc, self.PC, insn)
+        try:
+            implementation(*insn.operands)
+            self._icount += 1
+        finally:
+            self._publish('did_execute_instruction', self._last_pc, self.PC, insn)
```

Comment:
But did it execute the instruction ?
This is a problem. If the instruction raises a Concretize(..) it will re-enter in all the child states. Then both will_execute_instruction and did_execute_instructions are going to be executed 3 times for a single instruction ?